tests/src/testApplicationFactory.cc

```diff
@@ -0,0 +1,114 @@
+#define BOOST_TEST_MODULE TestApplicationFactory
+// Only after defining the name include the unit test header.
+#include "ApplicationFactory.h"
+#include "ReferenceTestApplication.h"
+
+#include <boost/test/included/unit_test.hpp>
+
+using namespace boost::unit_test_framework;
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_CASE(testNoFactoryNoInstance) {
+  try {
+    (void)ApplicationBase::getInstance();
+    BOOST_ERROR("ApplicationBase::getInstance() did not throw as expected.");
+  }
+  catch(ChimeraTK::logic_error& e) {
+    std::cout << "Exception message debug printout: " << e.what() << std::endl;
+  }
+
+  try {
+    (void)ApplicationFactoryBase::getApplicationInstance();
+    BOOST_ERROR("ApplicationFactoryBase::getInstance() did not throw as expected.");
+  }
+  catch(ChimeraTK::logic_error& e) {
+    std::cout << "Exception message debug printout: " << e.what() << std::endl;
+  }
+}
+
+BOOST_AUTO_TEST_CASE(testFactory) {
+  ApplicationFactory<ReferenceTestApplication> appFactory;
+
+  auto* firstCall = &ApplicationBase::getInstance();
+
+  auto referenceAppPointer = dynamic_cast<ReferenceTestApplication*>(firstCall);
+  BOOST_TEST(referenceAppPointer != nullptr);
+
+  auto* secondCall = &ApplicationBase::getInstance();
+
+  BOOST_TEST(firstCall == secondCall);
+}
+
+class AppWithParams : public ReferenceTestApplication {
+ public:
+  int a;
+  float b;
+  AppWithParams(int a_, float b_) : a(a_), b(b_) {}
+};
+
+BOOST_AUTO_TEST_CASE(testWithParams) {
+  ApplicationFactory<AppWithParams> appFactory(3, 5.8);
+
+  auto* withParams = dynamic_cast<AppWithParams*>(&ApplicationBase::getInstance());
+  BOOST_TEST(withParams != nullptr);
+  BOOST_TEST(withParams->a = 3);
+  BOOST_TEST(withParams->b = 5.8);
```

Comment:
floating-point comparison but no tolerance used. Either add `tt::tolerance(0.001)` as a second argument to BOOST_TEST or move to a different type which has a well defined exact comparison. (Linters will warn about this!)

Reply:
Hm, I thought BOOST_TEST was doing this automatically. I did not know that I explicitly have to turn it on. Thanks for the hint.

Reply:
> I thought BOOST_TEST was doing this automatically

How could it, it does not know the tolerance I can accept.

However, this is not a test at all. I am using the assignment operator :-O. The linter should warm about that.

Reply:
fixed